ui/src/features/Editor/components/LeftPanel/index.tsx

```diff
@@ -28,46 +40,112 @@ const LeftPanel: React.FC<Props> = ({
   onNodesAdd,
   isMainWorkflow,
   hasReader,
+  // onNodeDoubleClick,
+  selected,
 }) => {
   const t = useT();
   const { workspaceId } = useParams({ strict: false });
   const [selectedTab, setSelectedTab] = useState<Tab | undefined>();
+  const { fitView, getZoom, zoomTo } = useReactFlow();
+  const [previousZoom, setPreviousZoom] = useState<number | undefined>(
+    undefined,
+  );
+  const [nodeId, setNodeId] = useState<string | undefined>(undefined);
 
-  const [_content, setContent] = useState("Admin Page");
+  useEffect(() => {
+    if (previousZoom !== undefined && selected === undefined) {
+      zoomTo(previousZoom, { duration: 400 });
+    }
+  }, [previousZoom, selected, zoomTo]);
+
+  useEffect(() => {
+    if (selected && nodeId) {
+      setPreviousZoom(getZoom());
+    }
+
+    if (!isOpen) {
+      setNodeId(undefined);
+    }
+  }, [selected, isOpen, getZoom, nodeId]);
```

Comment:
I'm not sure this is working well. It zooms to the node, but doesn't return to the previous zoom. (I think actually there is no way to deselect, so that might be why).

Also, is there any way you can move this logic in to a callback that handles selection or something? Using useEffects in this way often causes headaches later.